https://issues.redhat.com/browse/ACM-25181 (#8339)
open-cluster-management/rhacm-docs: gitops/gitops_addon_without_argocd.adoc

[#enable-gitops-addon-without-argocd]
= Enabling {gitops} add-on without the _ArgoCD_ agent

> The basic pull model does not include the `ArgoCD` agent, so it gives you a simpler setup for your hub cluster management and only gives you the necessary statuses of the health of your hub clusters. 

Reviewer: please define "it"

To enable a {gitops-short} add-on without the `ArgoCD` agent, complete the following sections: 

* <<prerequisites, Prerequisites>>
* <<create-gitopscluster-resource, Creating a _GitOpsCluster_ resource>>
* <<verify-installation, Verifying the installation>>
* <<additional-resources, Additional resources>> 

> [#prerequisites]

Reviewer: ```suggestion
.Prerequisites
```

Author: I'm confused. You want me to remove the level header two for the prereq and write it like, `.Prerequisites`

so get rid of: 

[#prerequisites]
== Prerequisites

== Prerequisites 

> Ensure that you have the following resources installed: 

Reviewer: here, we don't **install** all of these, not sure why this is built like this with line 16 (?); I only meant to write them like we write the prereqs in the rest of the doc.

Reviewer: Please change line 16. Right now you say:

Ensure that you have the following resources installed:

{gitops-short} operator installed on the hub cluster

This doesn't agree bc you already say install. We don't need line 16 and it's not the most accurate for this list. The list alone is fine. :-)

> Ensure that you have the following resources installed: 

Reviewer: ```suggestion
See the following prerequisites for enablement:
```

* {acm-short} hub cluster installed

> * Managed clusters registered with {acm-short}

Reviewer: ```suggestion
* Managed clusters that are registered with {acm-short}
```

> * {gitops-short} operator installed on the hub cluster

Reviewer: ```suggestion
* {gitops-short} operator that is installed on the hub cluster
```

> * A `Placement` resource to select target managed clusters

Reviewer: ```suggestion
* A `Placement` resource that is defined to select target managed clusters
```

Reviewer: You can also say  _that specifies targeted managed clusters_, I think, based on what the resource does. From our doc, define may be better:

= Placement

A placement resource is a namespace-scoped resource that defines a rule to select a set of `ManagedClusters` from the `ManagedClusterSets`, which are bound to the placement namespace.

== Placement policy

A policy that defines where the application components are deployed and how many replicas there are.

> * `ManagedClusterSet` bound to the target namespace

Reviewer: ```suggestion
* `ManagedClusterSet` that is bound to the target namespace
```

== Creating a `GitOpsCluster` resource

To enable basic pull model, create a `GitOpsCluster` resource. The controller automatically creates the following resources for each managed cluster selected by the `Placement` policy: 

* `AddOnDeploymentConfig` resource in the managed cluster namespace
* `ManagedClusterAddOn` resource in the managed cluster namespace

The {gitops} add-on deploys to each selected managed cluster and installs the following resources:

* {gitops-short} operator in the `openshift-gitops-operator` namespace
* ArgoCD instance in the  `openshift-gitops` namespace

To create a `GitOpsCluster` resource, complete the following steps:

. On your hub cluster, create a `GitOpsCluster` resource to enable the {gitops} add-on by applying the following YAML sample:

+
[source,yaml]
----
apiVersion: apps.open-cluster-management.io/v1beta1
kind: GitOpsCluster
metadata:
  name: gitops-clusters
  namespace: openshift-gitops
spec:
  argoServer:
    argoNamespace: openshift-gitops
  placementRef:
    kind: Placement
    apiVersion: cluster.open-cluster-management.io/v1beta1
    name: all-openshift-clusters
    namespace: openshift-gitops
  gitopsAddon:
    enabled: true
----

> 

Reviewer: they created it, now did they apply it?

Author: @mikeshng I believe what we want users to do is in the step we provide: create the GitOpsCluster resource BY applying the yaml. 

If more steps are needed, please let me know.

Reviewer: there is usually an apply command that should be included in any procedure that calls to create and apply YAML. 

Honestly these should probably be: Step 1: Create Step 2: Apply 

We have looked at how to build YAML procedures in the doc (as a team) bc not everyone was doing that, but new doc should be doing that.

Reviewer 2: > I believe what we want users to do is in the step we provide: create the GitOpsCluster resource BY applying the yaml.
> If more steps are needed, please let me know.

Yes, we want the user to create the resource by applying the yaml. There are many ways to apply the yaml but the usual way is first create a file with the yaml content then run `oc apply -f yaml_file_name`

> Honestly these should probably be: Step 1: Create Step 2: Apply

Sounds good to me.

> ----

Reviewer: missing how to apply

[#verify-installation]
== Verifying the installation 

To verify the necessary installations and resources for successful deployment, complete the following steps:

. Verify that the `GitOpsCluster` resource has a status for successful deployment by running the following command:  

+
[source,bash]
----
oc get gitopscluster gitops-clusters -n openshift-gitops -o yaml
----

. Verify that the {gitops-short} add-on controller is working by running the following command: 

+
[source,bash]
----
oc get pods -n open-cluster-management-agent-addon
----

. Verify the {gitops-short} operator is working by running the following command: 

+
[source,bash]
----
oc get pods -n openshift-gitops-operator
----

. Verify the `ArgoCD`` instance is working by running the following command:

+
[source,bash]
----
oc get pods -n openshift-gitops
----

> 

Reviewer: this looks good!

== Creating a `GitOpsCluster` resource

To enable basic pull model, create a `GitOpsCluster` resource. The controller automatically creates the following resources for each managed cluster selected by the `Placement` policy: 

* `AddOnDeploymentConfig` resource in the managed cluster namespace
* `ManagedClusterAddOn` resource in the managed cluster namespace

The {gitops} add-on deploys to each selected managed cluster and installs the following resources:

* {gitops-short} operator in the `openshift-gitops-operator` namespace
* ArgoCD instance in the  `openshift-gitops` namespace

To create a `GitOpsCluster` resource, complete the following steps:

. On your hub cluster, create a `GitOpsCluster` resource to enable the {gitops} add-on by applying the following YAML sample:

+
[source,yaml]
----
apiVersion: apps.open-cluster-management.io/v1beta1
kind: GitOpsCluster
metadata:
  name: gitops-clusters
  namespace: openshift-gitops
spec:
  argoServer:
    argoNamespace: openshift-gitops
  placementRef:
    kind: Placement
    apiVersion: cluster.open-cluster-management.io/v1beta1
    name: all-openshift-clusters
    namespace: openshift-gitops
  gitopsAddon:
    enabled: true
----

> . Finish applying and save the YAML sample by running the following command: 

Reviewer: Please change all of these to:

**. Run the following command to apply your YAML sample:**

We don't need this many words. 

I realize you say _apply_ earlier (we mostly just say _add_ or _create_ in those first steps, but it's not incorrect.) The command is how you actually "apply" and the previous step is creating or adding to the YAML.

[#enable-gitops-addon-without-argocd]
= Enabling {gitops} add-on without the _ArgoCD_ agent

> The `Basic` pull model does not include the `ArgoCD` agent, so it gives you a simpler setup for your hub cluster management and only gives you the necessary statuses of the health of your hub clusters. 

Reviewer: Why is Basic coded?

Reviewer: Same with the GitOps name throughout, see where the conref needs to be used.

To enable a GitOps add-on without the `ArgoCD` agent, complete the following sections: 

* <<prerequisites, Prerequisites>>
* <<create-gitopscluster-resource, Creating a _GitOpsCluster_ resource>>
* <<verify-installation, Verifying the installation>>
* <<additional-resources, Additional resources>> 

[#prerequisites]

> == Prerequisites 

Reviewer: Same feedback. You tell them to "complete" this section in the line before, so I would go with different language, but either way these are complete sentences in most of the books. 

Ensure you have an {acm-short} hub cluster installed. 
Register your
Install OCP GitOps operator
etc...

* {acm-short} hub cluster installed
* Managed clusters registered with {acm-short}
* OpenShift GitOps operator installed on the hub cluster
* A `Placement` resource to select target managed clusters
* `ManagedClusterSet` bound to the target namespace

[#create-gitopscluster-resource]
== Creating a `GitOpsCluster` resource

> Instead of using an `ArgoCD` agent, you can install a `GitOpsCluster` resource. After you install a `GitOpsCluster` resource, its controller automatically creates the following resources for each managed cluster selected by the `Placement` policy: 

Reviewer: same with possessive. I usually just say "the"

> Instead of using an `ArgoCD` agent, you can install a `GitOpsCluster` resource. After you install a `GitOpsCluster` resource, its controller automatically creates the following resources for each managed cluster selected by the `Placement` policy: 

Reviewer 2: ```suggestion
To enable basic pull model, create a `GitOpsCluster` resource, its controller automatically creates the following resources for each managed cluster selected by the `Placement` policy: 
```

> *  `AddOnDeploymentConfig` in the managed cluster namespace

Reviewer: extra space

Reviewer: ```suggestion
* `AddOnDeploymentConfig` resource in the managed cluster namespace
```

> * `ManagedClusterAddOn` in the managed cluster namespace

Reviewer: ```suggestion
* `ManagedClusterAddOn` resource in the managed cluster namespace
```

The {gitops} add-on deploys to each selected managed cluster and installs the following resources:

* OpenShift GitOps operator in the `openshift-gitops-operator` namespace

> * Argo CD instance in the  `openshift-gitops` namespace

Reviewer 2: ```suggestion
* ArgoCD instance in the  `openshift-gitops` namespace
```

To create a `GitOpsCluster` resource, complete the following steps:

> . On your hub cluster, create a `GitOpsCluster` resource to enable the {gitops} add-on:

Reviewer: See the following...

+
[source,yaml]
----
apiVersion: apps.open-cluster-management.io/v1beta1
kind: GitOpsCluster
metadata:
  name: gitops-clusters
  namespace: openshift-gitops
spec:
  argoServer:
    argoNamespace: openshift-gitops
  placementRef:
    kind: Placement
    apiVersion: cluster.open-cluster-management.io/v1beta1
    name: all-openshift-clusters
    namespace: openshift-gitops
  gitopsAddon:
    enabled: true
----

[#verify-installation]
== Verifying the installation 

To verify the necessary installations and resources for successful deployment, complete the following steps:

. Verify that the `GitOpsCluster` resource has a status for successful deployment by running the following command:  

+
[source,bash]
----
oc get gitopscluster gitops-clusters -n openshift-gitops -o yaml
----

> . Verify that  the GitOps add-on controller is working by running the following command: 

Reviewer: extra space

Reviewer 2: This command and the following commands are meant to run on the managed cluster (where the addon is installed).

+
[source,bash]
----
oc get pods -n open-cluster-management-agent-addon
----

. Verify the OpenShift GitOps operator is working by running the following command: 

+
[source,bash]
----
oc get pods -n openshift-gitops-operator
----

. Verify the Argo CD instance is working by running the following command:

+
[source,bash]
----
oc get pods -n openshift-gitops
----

[#additional-resources]
== Additional resources 

> Continue learning about the {gitops-short} add-on by completing xref:../gitops/gitops_manage_addon.adoc#manage-gitops-addon[Managing the {gitops} add-on].

Reviewer: same with "learning" but you are sending them to a task to "complete"
